Homework/src/main/java/io/codelex/advancedtest/exercise1/CreditCard.java

```diff
@@ -0,0 +1,25 @@
+package io.codelex.advancedtest.exercise1;
+
+import java.math.BigDecimal;
+
+public class CreditCard extends Card {
+    public CreditCard(String cardNumber, String owner, int ccv, double balance) {
+        super(cardNumber, owner, ccv, balance);
+    }
+
+    @Override
+    public void takeMoney(double money) {
+        int test = super.getBalance().compareTo(BigDecimal.valueOf(money));
```

Comment:
Redzēju, ka šo mainīji no `this` uz `super`. Faktiski te var lietot bez neviena - vienkārši `getBalance()`, jo šajā klasē nav pārrakstīta getBalance() - attiecīgi vienīgā getBalance metode ir no parent klases.